trajopt/src/problem_description.cpp

```diff
@@ -777,12 +777,24 @@ void DynamicCartPoseTermInfo::hatch(TrajOptProb& prob)
   }
   else
   {
-    auto f = std::make_shared<DynamicCartPoseErrCalculator>(
-        prob.GetKin(), source_frame, target_frame, source_frame_offset, target_frame_offset, indices);
+    auto f = std::make_shared<DynamicCartPoseErrCalculator>(prob.GetKin(),
+                                                            source_frame,
+                                                            target_frame,
+                                                            source_frame_offset,
+                                                            target_frame_offset,
+                                                            indices,
+                                                            lower_tolerance,
+                                                            upper_tolerance);
 
     // This is currently not being used. There is an intermittent bug that needs to be tracked down it is not used.
-    auto dfdx = std::make_shared<DynamicCartPoseJacCalculator>(
-        prob.GetKin(), source_frame, target_frame, source_frame_offset, target_frame_offset, indices);
+    auto dfdx = std::make_shared<DynamicCartPoseJacCalculator>(prob.GetKin(),
```

Comment:
The jacobian should not take into account tolerances.

Reply:
Makes sense, and I'll remove that, but this appears to be not used at all anyway.

Reply:
Yea, but for this to work we will need to implement or update these existing ones to correctly compute the jacobian.

Reply:
Removed all the tolerances from the Jacobians, but haven't made it so the Jacobians are actually used yet.

Reply:
You seem to say the Jacobian issue is resolved in #139, but I don't see how/where this was resolved.

Reply:
Maybe it was fixed in #266 but no one ever changed `problem_description.cpp` to reflect this.

Reply:
In my testing my plans always fail if I pass dfdx into the cost and/or constraint. The changes to `kinematic_terms.cpp` in #266 do not seem to have resolved the issue. @Levi-Armstrong, any ideas on what's wrong [here](https://github.com/tesseract-robotics/trajopt/blob/806a40a7b48ac21bbee935f1f8eac3974d2385c6/trajopt/src/kinematic_terms.cpp#L158)?

Reply:
We cannot use the existing one because there is a known issue which needs to be resolved. We need to do the same thing that the current jacobian from error function does but use the error function which does not take tolerances into account.